### PR TITLE
Refactor the pipeline into smaller parts for easier integration.

### PR DESCRIPTION
**Pull request by rhpvorderman**
I wanted to be able to use a single-sample pipeline. So now I have refactored the pipeline to be imported into three parts
- calculate regions. Where you calculate the regions (in case you want to call with X-ploidy aware stuff). It also calculates the scatters.
- Do single sample calling. With given scatters.
- Do joint genotyping (optional, single sample has to be set to gvcf mode).

### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Documentation was updated (if required).

### CODE REVIEW

File: calculate-regions.wdl [hunk 1]
```diff
@@ -20,72 +20,110 @@ version 1.0
...
+        File? Xregions = if defined(regions) then intersectX.intersectedBed else XNonParRegions
+        File? Yregions = if defined(regions) then intersectY.intersectedBed else YNonParRegions
+        File? autosomalRegions = calculatedAutosomalRegions
+        Array[File] autosomalRegionScatters = scatterAutosomalRegions.scatters
     }
 
     parameter_meta {
         regions: {description: "A bed file describing the regions to operate on. Will be used to intersect the other regions.", category: "common"}
         XNonParRegions: {description: "Bed file with the non-PAR regions of X", category: "required"}
         YNonParRegions: {description: "Bed file with the non-PAR regions of Y", category: "required"}
```
Comment (DavyCats) [inline calculate-regions.wdl]:
```suggestion
        XNonParRegions: {description: "Bed file with the non-PAR regions of X", category: "common"}
        YNonParRegions: {description: "Bed file with the non-PAR regions of Y", category: "common"}
```

File: docs/index.md [hunk 2]
```diff
@@ -14,7 +14,20 @@ https://www.lumc.nl/).
...
+up into three parts for convenience.
+
++ `calculate-regions.wdl`: Calculates the scatters taking into account 
+  (optional) the non-PAR regions and regions of interest.
++ `single-sample-variantcalling.wdl`: Variant calling using the given scatters
+  and non-PAR regions. If no scattering is used and there are no non-PAR 
+  regions given, this is the same as a plain call to HaplotypeCaller. Can be
+  run in GVCF mode.
++ `jointgenotyping.wdl`. Runs the jointgenotyping (scattered). This step is
+  optional. Requires the single sample pipeline to be run in GVCF mode.
```
Comment (DavyCats) [inline docs/index.md:30]: Shouldn't this be outside of the codeblock?

File: jointgenotyping.wdl [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    output {
+        File multisampleGVcf = gatherGvcfs.outputVcf
+        File multisampleGVcfIndex = gatherGvcfs.outputVcfIndex
+        File multisampleVcf = select_first([noScatterVcf, gatherVcfs.outputVcf])
+        File multisampleVcfIndex = select_first([noScatterVcfIndex, gatherVcfs.outputVcfIndex])
+    }
+
+    parameter_meta {
+        gvcfFiles: {description: "List of GVCF files to merge and genotype jointly.", category: "required"}
+        gvcfFilesIndex: {description: "The indexes for the GVCF files."}
```
Comment (DavyCats) [inline jointgenotyping.wdl]: missing category

File: single-sample-variantcalling.wdl [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        File? outputVcf = if scattered then mergedVcf else noScatterVcf
+        File? outputVcfIndex = if scattered then mergedVcfIndex else noScatterVcfIndex
+        Array[File] vcfScatters = VCFs
+        Array[File] vcfIndexScatters = VCFIndexes
+    }
+
+    parameter_meta {
+        bam: {description: "The bam file", category: "required"}
+        bamIndex: {description: "Index of the bam file", category: "required"}
+        gender: {description: "Gender of the sample. Accepted values: female,F,f,male,M,m. Other values default to 'unknown'."}
```
Comment (DavyCats) [inline single-sample-variantcalling.wdl]: missing category

**Review by DavyCats — CHANGES_REQUESTED** [review]

**Comment by rhpvorderman** [comment]
@DavyCats Thanks! Issues have been resolved.

**Review by DavyCats — APPROVED** [review]

**Pull request merged**